powerauth-java-cmd-lib/src/main/java/io/getlime/security/powerauth/lib/cmd/steps/AbstractBaseStep.java

```diff
@@ -179,44 +196,65 @@ public final JSONObject execute(StepLogger stepLogger, Map<String, Object> conte
     }
 
     /**
-     * Prepares ECIES encryptor and encrypts request data with sharedInfo1.
+     * Prepares encryptor and encrypts request data with given encryptor.
      * The encrypted request is then added to the request context of this step.
      *
      * @param stepContext       Context of this step
+     * @param applicationKey    Application key.
      * @param applicationSecret Application secret
-     * @param eciesSharedInfo   Parameter sharedInfo1
+     * @param encryptorId       Encryptor identifier
      * @param data              Request data for the encryption
-     * @param associatedData    Data associated with ECIES
      * @throws Exception when an error during encryption of the request data occurred
      */
-    public void addEncryptedRequest(StepContext<M, R> stepContext, String applicationSecret, EciesSharedInfo1 eciesSharedInfo, byte[] data, byte[] associatedData) throws Exception {
+    public void addEncryptedRequest(StepContext<M, R> stepContext, String applicationKey, String applicationSecret, EncryptorId encryptorId, byte[] data) throws Exception {
         M model = stepContext.getModel();
         SimpleSecurityContext securityContext = (SimpleSecurityContext) stepContext.getSecurityContext();
         ResultStatusObject resultStatusObject = model.getResultStatus();
 
-        EciesEncryptor encryptor;
-        EciesParameters parameters;
+        ClientEncryptor encryptor;
         if (securityContext == null) {
-            parameters = getRequestEciesParameters(stepContext.getModel().getVersion(), associatedData);
-            encryptor = SecurityUtil.createEncryptorForActivationScope(applicationSecret, resultStatusObject, eciesSharedInfo, parameters);
+            final byte[] transportMasterKeyBytes = Base64.getDecoder().decode(resultStatusObject.getTransportMasterKey());
+            encryptor = ENCRYPTOR_FACTORY.getClientEncryptor(
```

Comment:
This code is a bit hard to read, would you mind splitting it a bit? Also the formatting is off for `applicationKey`.